libsoftwarecontainer/src/gateway/network/networkgatewayparser.cpp

```diff
@@ -162,19 +134,42 @@ ReturnCode NetworkGatewayParser::parsePort(const json_t *element,  IPTableEntry:
     return ReturnCode::SUCCESS;
 }
 
-IPTableEntry::Target NetworkGatewayParser::parseTarget(const std::string &str)
+ReturnCode NetworkGatewayParser::parseProtocol(const json_t *element,
+                                               std::vector<std::string> &proto)
 {
-    if (str == "ACCEPT") {
-        return IPTableEntry::Target::ACCEPT;
-    }
+    // Single Protocol
+    if (json_is_string(element)) {
+        std::string protocol = json_string_value(element);
+        if ((protocol !=  "tcp") &&  (protocol !=  "udp") && (protocol !=  "icmp")) {
+            log_error() << protocol
+                        << " is not appropriate only tcp, udp and icmp protocol are alllowed";
+            return ReturnCode::FAILURE;
+        }
+        proto.push_back(protocol);
+    // Multiple protocols
+    } else if (json_is_array(element)) {
+        size_t ix;
+        json_t *val;
 
-    if (str == "DROP") {
-        return IPTableEntry::Target::DROP;
-    }
+        json_array_foreach(element, ix, val) {
+            if (!json_is_string(val)) {
+                log_error() << "Inappropriate protocol is listeed";
+                return ReturnCode::FAILURE;
+            }
 
-    if (str == "REJECT") {
-        return IPTableEntry::Target::REJECT;
-    }
+            std::string protocol = json_string_value(val);
 
-    return IPTableEntry::Target::INVALID_TARGET;
- }
+            if ((protocol !=  "tcp") &&  (protocol !=  "udp") && (protocol !=  "icmp")) {
+                log_error() << protocol
+                            << " is not appropriate only tcp, udp and icmp protocol are alllowed";
+                return ReturnCode::FAILURE;
```

Comment:
Couldn't you break out the protocol parsing into a metthod, since you're doing the same thing for a single and for an array of protocols?